src/appshell/internal/startupscenario.cpp

```diff
@@ -327,7 +327,8 @@ void StartupScenario::onStartupPageOpened(StartupModeType modeType)
         configuration()->setWelcomeDialogLastShownIndex(-1); // reset
     }
 
-    if (configuration()->welcomeDialogShowOnStartup() && !museSoundsUpdateScenario()->hasUpdate()) {
+    const int numInstances = multiInstancesProvider()->instances().size();
```

Comment:
I'm afraid this is going to give a compiler warning on Windows about int vs size_t